src/curated-corpus/pages/ProspectingPage/ProspectingPage.tsx

```diff
@@ -345,37 +345,66 @@ export const ProspectingPage: React.FC = (): JSX.Element => {
       { variables: { data: { ...approvedItem } } },
       'Item successfully added to the curated corpus.',
       (approvedItemData) => {
-        // call the mutation to mark prospect as approved
-        runMutation(
-          updateProspectAsCurated,
-          { variables: { prospectId: currentProspect?.id } },
-          undefined,
-          () => {
-            toggleApprovedItemModal();
-
-            if (approvedItem.status === CuratedStatus.Recommendation) {
-              setApprovedItem(approvedItemData.createApprovedCuratedCorpusItem);
-              toggleScheduleModal();
+        // if we have a prospect id, we need to tell prospect api that this
+        // prospect has been curated
+        if (currentProspect?.id !== '') {
+          // call the mutation to mark prospect as approved
+          runMutation(
+            updateProspectAsCurated,
+            { variables: { prospectId: currentProspect?.id } },
+            undefined,
+            () => {
+              postCreateApprovedItem(
+                approvedItem.status,
+                approvedItemData,
+                true
+              );
+
+              formikHelpers.setSubmitting(false);
+            },
+            () => {
+              formikHelpers.setSubmitting(false);
             }
+          );
+        } else {
+          // if we don't have a prospect id, this was a manually added prospect
+          // and we don't need to call prospect api at all
+          postCreateApprovedItem(approvedItem.status, approvedItemData, false);
 
-            // Remove the newly curated item from the list of prospects displayed
-            // on the page.
-            setProspects(
-              prospects.filter(
-                (prospect) => prospect.id !== currentProspect?.id!
-              )
-            );
-
-            formikHelpers.setSubmitting(false);
-          },
-          () => {
-            formikHelpers.setSubmitting(false);
-          }
-        );
+          formikHelpers.setSubmitting(false);
+        }
       }
     );
   };
 
+  /**
+   * common stuff we (may) need to do after creating an approved item
+   *
+   * @param approvedItemStatus (CuratedStatus): the status of the approved item
+   * @param approvedItemData (any): the data we get back from the createApprovedItem mutation
+   * @param filterProspects (boolean): whether or not we need to filter the list of prospects on the screen
+   */
+  const postCreateApprovedItem = (
+    approvedItemStatus: CuratedStatus,
+    approvedItemData: any,
```

Comment:
is there a better type i can give this? i couldn't figure out how to tell what this is exactly.